### PR TITLE
[4.1] Fileinfo check

### PR DESCRIPTION
**Pull request by brianteeman**
Despite the php extension fileinfo being enabled by default in php we have had quite a few users who are discovering that media manager will not load. This is because they have fileinfo disabled on the server.

This PR adds fileinfo to the list of php settings shown in the system information.

To test
Install the PR and check the php settings page of the system information and you will see the new line "Fileinfo Available" which hopefully has a setting of yes.

In your php.ini file locate the line with `extension=fileinfo` and comment it out by placing at ; at the beginning of the line.

Then restart apache

Now when you go to media manager you will see the reported error.

image

Check the php settings page of the system information and you will see that fileinfo is not available

image

Now go back to php.ini and uncomment that line

Restart apache

And check that media manager works and the system setting shows fileinfo is available

image

### CODE REVIEW

File: administrator/components/com_admin/tmpl/sysinfo/default_phpsettings.php [hunk 1]
```diff
@@ -143,6 +143,14 @@
 					<?php echo HTMLHelper::_('phpsetting.string', $this->phpSettings['disable_functions']); ?>
 				</td>
 			</tr>
+			<tr>
+				<th scope="row">
+				<?php echo Text::sprintf('COM_ADMIN_EXTENSION_AVAILABLE', 'Fileinfo'); ?>
```
Comment (Quy) [inline administrator/components/com_admin/tmpl/sysinfo/default_phpsettings.php]:
```suggestion
					<?php echo Text::sprintf('COM_ADMIN_EXTENSION_AVAILABLE', 'Fileinfo'); ?>
```

File: administrator/components/com_admin/tmpl/sysinfo/default_phpsettings.php [hunk 2]
```diff
@@ -143,6 +143,14 @@
...
+				</th>
+				<td class="break-word">
```
Comment (Quy) [inline administrator/components/com_admin/tmpl/sysinfo/default_phpsettings.php]: Is `break-word` necessary?

Comment (brianteeman) [reply administrator/components/com_admin/tmpl/sysinfo/default_phpsettings.php]: probably not. it was a copy paste. will remove it

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 06ef0e3c45a45aae62c16ecd9678c43f2137e242<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37081">issues.joomla.org/tracker/joomla-cms/37081</a>.</sub>

**Comment by zero-24** [comment]
So its kind of a soft requirement? IIRC we do have an optional php modules list on docs where we can add this too when its not there yet right?

**Comment by brianteeman** [comment]
> So its kind of a soft requirement? 

Soft requirement for Joomla 
Hard requirement for Media Manager

> IIRC we do have an optional php modules list on docs where we can add this too when its not there yet right?

Possibly - no idea

**Comment by zero-24** [comment]
Its this page: https://docs.joomla.org/J4.x:Optional_Technical_Requirements

> Soft requirement for Joomla
> Hard requirement for Media Manager

Agree

**Comment by brianteeman** [comment]
Documentation updated 
image

**Comment by zero-24** [comment]
Thank you Brian!

**Comment by Quiviro** [comment]
I have tested this item :white_check_mark: successfully on 06ef0e3c45a45aae62c16ecd9678c43f2137e242<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37081">issues.joomla.org/tracker/joomla-cms/37081</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37081">issues.joomla.org/tracker/joomla-cms/37081</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Comment by bembelimen** [comment]
@zero-24 something we should add in the pre-install checker of 3.10?

**Comment by brianteeman** [comment]
Thank you

**Comment by zero-24** [comment]
> @zero-24 something we should add in the pre-install checker of 3.10?

Hmm its an optional requirement you could theoretical replace the core media manager with one that does not rely on this module so I'm not sure whether we should add it in there.

**Comment by roland-d** [comment]
How many users really do that? We could at least give users a heads up. We know for a fact that the media manager will not work without fileinfo

**Pull request merged**